client/app/src/androidTest/java/com/example/criengine/ViewMyBookTest.java

```diff
@@ -107,22 +100,48 @@ public void cancelEditTitleTest() {
      */
     @Test
     public void cancelDeleteBookTest() {
-        assertTrue(solo.waitForText("Delete Book", 1, 2000));
         solo.clickOnButton("Delete Book");
 
         assertTrue(solo.waitForText("Are you sure you want to delete this book?", 1, 2000));
         solo.clickOnText("Cancel");
 
         // Make sure we didn't change screens.
         solo.assertCurrentActivity("Wrong Activity", MyBookActivity.class);
+
+        solo.goBack();
+        cleanup();
+    }
+
+    /**
+     * Add a mock book.
+     */
+    public void addBook() {
```

Comment:
If these are all the same, you can make it into a static method of another class that does this for all the cases, similarly for cleanup (especially if it's not before and after all the tests)

Reply:
Sure can do